rewrite-maven/src/test/java/org/openrewrite/maven/cleanup/RemoveCompileScopeTest.java

```diff
@@ -0,0 +1,214 @@
+                    <version>6.0.5</version>
+                  </dependency>
+                </dependencies>
+              </project>
+              """
+          )
+        );
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"provided","runtime","test","system"})
+    void otherScopesShouldNotBeTouched(String scope) {
+        rewriteRun(
+          pomXml(
+            //language=xml
+            """
+              <project>
+                <modelVersion>4.0.0</modelVersion>
+                
+                <groupId>com.mycompany.app</groupId>
+                <artifactId>my-app</artifactId>
+                <version>1</version>
+                
+                <dependencies>
+                  <dependency>
+                    <groupId>com.google.guava</groupId>
+                    <artifactId>guava</artifactId>
+                    <version>28.2-jre</version>
+                    <scope>%s</scope>
+                  </dependency>
+                </dependencies>
+              </project>
+              """.formatted(scope)
+          )
+        );
+    }
+
+    @Test
+    void missingScopeShouldNotBeTouched() {
+        rewriteRun(
+          pomXml(
+            """
+              <project>
+                <modelVersion>4.0.0</modelVersion>
+                
+                <groupId>com.mycompany.app</groupId>
+                <artifactId>my-app</artifactId>
+                <version>1</version>
+                
+                <dependencies>
+                  <dependency>
+                    <groupId>com.google.guava</groupId>
+                    <artifactId>guava</artifactId>
+                    <version>28.2-jre</version>
+                  </dependency>
+                </dependencies>
+              </project>
+              """
+          )
+        );
+    }
+
+    @Test
+    void testDependenciesManagementIsUntouched() {
+        rewriteRun(
+          pomXml(
+            """
+              <project>
+                <modelVersion>4.0.0</modelVersion>
+                
+                <groupId>com.mycompany.app</groupId>
+                <artifactId>my-app</artifactId>
+                <version>1</version>
+                <dependencyManagement>
+                    <dependencies>
+                        <dependency>
+                            <groupId>group-a</groupId>
+                            <artifactId>artifact-a</artifactId>
+                            <version>1.0</version>
+                        </dependency>
+                        <dependency>
+                            <groupId>com.google.guava</groupId>
+                            <artifactId>guava</artifactId>
+                            <version>28.2-jre</version>
+                            <scope>compile</scope>
+                        </dependency>
+                    </dependencies>
+                </dependencyManagement>
+              </project>
+              """
+          )
+        );
+    }
```

Comment:
Might want to add a test case here where you override the scope from dependencyManagement, if I [recall correctly](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-scope). And note that the dependency management might also be in a parent, so that makes this a bit harder to determine perhaps. We would not want to remove an overridden scope if that changes the behaviour.

```suggestion
    }

    @Test
    void overridenDependencyManagementScopeIsUntouched() {
        rewriteRun(
          pomXml(
            """
              <project>
                <modelVersion>4.0.0</modelVersion>

                <groupId>com.mycompany.app</groupId>
                <artifactId>my-app</artifactId>
                <version>1</version>
                <dependencyManagement>
                    <dependencies>
                        <dependency>
                            <groupId>com.google.guava</groupId>
                            <artifactId>guava</artifactId>
                            <version>28.2-jre</version>
                            <scope>test</scope>
                        </dependency>
                    </dependencies>
                </dependencyManagement>
                <dependencies>
                    <dependency>
                        <groupId>com.google.guava</groupId>
                        <artifactId>guava</artifactId>
                        <version>28.2-jre</version>
                        <scope>compile</scope>
                    </dependency>
                </dependencies>
              </project>
              """
          )
        );
    }
```

Reply:
Thank you for the insight. This is another example where to make a decision I'd need to have the entire document in front of me beforehand 

9f77267c93d6ffa5ba27f6f9f81ffb69a1393d97 is a decent attempt but that does not work when the order is different (see the new test added)

Do you have any examples where I can visit the document twice?  I think that using this approach might work, but I'd need to scan the entire document and then run it again with the filled `seen` entries

if there is any better way to do this please let me know

Reply:
@yeikel You can use `visit(List<SourceFile> before, ExecutionContext ctx)` instead of `getVisitor()` in a recipe. In many cases we take multiple full passes over a file (or a whole project) in the implementation of that method.

Reply:
Thank you for the help @jkschneider 

I pushed d08c82cb35c5c56a2e830bf926cb54aea2bbe133 using that idea